msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java

```diff
@@ -3045,40 +3045,38 @@ public void setName(String name) {
 	
 	public Map<String, Integer> getStudentTopicMessagCount(DiscussionForum forum, DiscussionTopic currTopic, Integer topicTotalCount, Map<String, Boolean> overridingPermissionMap){
 		Map<String, Integer> studentTotalCount = new HashMap<String, Integer>();
-		for(Entry<String, Boolean> entry: overridingPermissionMap.entrySet()){
-
-			if ((!forum.getDraft() && currTopic.getDraft().equals(Boolean.FALSE))
-					|| entry.getValue())
-			{ // this is the start of the big topic if
-				// set the message count for moderated topics, otherwise it will be set later
-				if(uiPermissionsManager.isRead(currTopic, (DiscussionForum)currTopic.getOpenForum(), entry.getKey())){
```

Comment:
I believe this check (inside the user loop) is saying: does the current instructor user have ability to view this topic?

Yes, it passes the user id to isRead (entry.getKey()), but then isRead seems to ignore that userId and just finds the current user? Crazy right?